File: src/main/java/io/github/oleksivio/telegram/bot/core/model/method/NetworkErrorAction.java
package io.github.oleksivio.telegram.bot.core.model.method;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.github.oleksivio.telegram.bot.api.model.NetworkError;

import java.util.Optional;

public abstract class NetworkErrorAction<RES> extends Action<RES> {

    @JsonIgnore
    private NetworkError networkError = null;

    protected void pSetNetworkErrorListener(NetworkError onNetworkError) {
        networkError = onNetworkError;
    }

    @JsonIgnore
    Optional<NetworkError> getNetworkErrorListener() {
        return Optional.ofNullable(networkError);
    }

    @JsonIgnore
    public abstract NetworkErrorAction<RES> setNetworkErrorListener(NetworkError onNetworkError);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/model/method/NetworkErrorAction.kt
package io.github.oleksivio.telegram.bot.core.model.method

import com.fasterxml.jackson.annotation.JsonIgnore
import io.github.oleksivio.telegram.bot.api.model.NetworkError

abstract class NetworkErrorAction<RES> : Action<RES>() {
    @JsonIgnore
    var networkErrorListener: NetworkError? = null
}
